### DISCUSSION TITLE
Example for Form AutoComplete Broken

### DISCUSSION THREAD

**Discussion by gjcarter2** [primefaces/community#796] [PrimeFaces] [answered]
I was working through some of the examples and found that the AutoComplete in the Forms section seems to be missing code.

Is this something that will be fixed at some point or is the example broken?

https://www.primefaces.org/showcase/ui/input/autoComplete.xhtml?jfwid=59ecd

**Comment by melloware** [comment]
Can you elaborate on what code is missing?

**Reply by gjcarter2** [reply]
AutoCompleteView.ajava:
 LazyCustomerDataModel definition isn't defined.
CustomerService.java:
 SecureRandom definition isn't defined.
 Representative definition isn't defined.
Customer.java:
 CustomerStatus definition isn't defined.

Thanks!

**Reply by melloware** [reply, marked as answer]
I will fix the showcase but of course all the source is available: https://github.com/primefaces/primefaces/tree/master/primefaces-showcase

**Reply by melloware** [reply]
Also `SecureRandom` is a Java class `import java.security.SecureRandom;`

**Reply by gjcarter2** [reply]
So I see. For some reason Eclipse doesn't see it and did not suggest it as being available to import.  (java.security.SecureRandom).

Mighty need to file an Eclipse bug.

**Reply by melloware** [reply]
@gjcarter2 are you sure you are using Java 8 or higher?  https://docs.oracle.com/javase/8/docs/api/java/security/SecureRandom.html

**Comment by gjcarter2** [comment]
Yes.  It's an Eclipse bug because without making any IDE settings changes
if I type it in manually it finds the class.

If I try to wait for it to auto complete my import or suggest an import in
the editor it can't find it.

**Comment by gjcarter2** [comment]
> Also `SecureRandom` is a Java class `import java.security.SecureRandom;`



> I was working through some of the examples and found that the AutoComplete in the Forms section seems to be missing code.
> 
> Is this something that will be fixed at some point or is the example broken?
> 
> https://www.primefaces.org/showcase/ui/input/autoComplete.xhtml?jfwid=59ecd

Yes, I see that the Web page for the demos is simply missing the tabs for certain pieces of code that is not included.

I found the definition for example of the Representitive here:

https://github.com/primefaces/primefaces/blob/master/primefaces-showcase/src/main/java/org/primefaces/showcase/domain/Representative.java

I suspect people new to Front End Java programming like myself (always worked on the server side never the browser side) would find this situation sort of confusing.

Some of the examples have all the definitions, while some like above are missing definitions for object references in the GUI example. Since I and others are new, I wasn't sure if the definition was some sort of template structure or definition required specifically for the PrimeFaces library or if it is just part of a hypothetical example.  For myself I though I thought I just didn't setup my IDE properly or I missed something in the IDE settings.

So might I suggest, you insure the examples have all the proper tabs for the source code?  I am going through most of the examples to learn how to use the GUI, so I can send you stuff I find on the example web pages that although maybe in the git source code tree, do not appear anywhere and may be a point of confusion for the newbie.

So far though I am enjoying the GUI/Front End programming though. Since browsers have matured so much, you can actually do desktop apps in the browser now so interesting possibilities are there.

**Reply by melloware** [reply]
Yep I already fixed it for next deploy: https://github.com/primefaces/primefaces/pull/11219/files

**Comment by gjcarter2** [comment]
Fabulous.  I may drop you a note if I find others.  Otherwise I will look
for missing items in the GIT tree if I can't find it from the example web
page.

**Reply by melloware** [reply]
New Showcase is deployed
